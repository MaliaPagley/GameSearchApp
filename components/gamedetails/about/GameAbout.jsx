import React from 'react'
import { View, Text } from 'react-native'

import styles from './gameabout.style'

const GameAbout = ({ description }) => {
  return (
    <View>
      <Text style={styles.pageHeaders}>About:</Text>
      <View style={styles.gameInfoBox} horizontal>
        <Text style={styles.gameInfo}>{description}</Text>
      </View>
    </View>
  )
}

export default GameAbout