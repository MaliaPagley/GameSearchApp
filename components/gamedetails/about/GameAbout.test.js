import React from 'react';
import { render } from '@testing-library/react-native';
import GameAbout from './GameAbout';

describe('GameAbout Component', () => {
  it('component renders', () => {
    const { toJSON } = render(<GameAbout />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders component with provided description', () => {
    const description = 'This is a test description.';
    const { getByText } = render(<GameAbout description={description} />);

    // Check if the component renders the header and the provided description
    expect(getByText('About:')).toBeTruthy();
    expect(getByText(description)).toBeTruthy();

    // Snapshot testing
    const tree = render(<GameAbout description={description} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders component without description', () => {
    const { getByText } = render(<GameAbout />);

    // Check if the component renders the header and the default "No description available." text
    expect(getByText('About:')).toBeTruthy();
    expect(getByText('No description available.')).toBeTruthy();

    // Snapshot testing
    const tree = render(<GameAbout />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
