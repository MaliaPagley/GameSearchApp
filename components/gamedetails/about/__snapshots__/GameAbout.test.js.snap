// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameAbout Component component renders 1`] = `
<View>
  <Text
    style={
      {
        "color": "#F3F4F8",
        "fontFamily": "dm-bold",
        "fontSize": 24,
        "marginLeft": 12,
        "marginTop": 16,
      }
    }
  >
    About:
  </Text>
  <View
    horizontal={true}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginLeft": 12,
        "marginTop": 6,
        "paddingTop": 6,
      }
    }
  >
    <Text
      style={
        {
          "color": "#F3F4F8",
        }
      }
    >
      No description available.
    </Text>
  </View>
</View>
`;

exports[`GameAbout Component renders component with provided description 1`] = `
<View>
  <Text
    style={
      {
        "color": "#F3F4F8",
        "fontFamily": "dm-bold",
        "fontSize": 24,
        "marginLeft": 12,
        "marginTop": 16,
      }
    }
  >
    About:
  </Text>
  <View
    horizontal={true}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginLeft": 12,
        "marginTop": 6,
        "paddingTop": 6,
      }
    }
  >
    <Text
      style={
        {
          "color": "#F3F4F8",
        }
      }
    >
      This is a test description.
    </Text>
  </View>
</View>
`;

exports[`GameAbout Component renders component without description 1`] = `
<View>
  <Text
    style={
      {
        "color": "#F3F4F8",
        "fontFamily": "dm-bold",
        "fontSize": 24,
        "marginLeft": 12,
        "marginTop": 16,
      }
    }
  >
    About:
  </Text>
  <View
    horizontal={true}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginLeft": 12,
        "marginTop": 6,
        "paddingTop": 6,
      }
    }
  >
    <Text
      style={
        {
          "color": "#F3F4F8",
        }
      }
    >
      No description available.
    </Text>
  </View>
</View>
`;
